src/inc/til/point.h

```diff
@@ -193,6 +196,19 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         {
             return { x, y };
         }
+
+        // til::point and POINT have the exact same layout at the time of writing,
+        // so this function lets you unsafely "view" this point as a POINT
+        // if you need to pass it to a Win32 function.
+        //
+        // Use as_win32_point() as sparingly as possible because it'll be a pain to hack
+        // it out of this code base once til::point and POINT aren't the same anymore.
+        // Prefer casting to POINT and back to til::point instead if possible.
+        POINT* as_win32_point() noexcept
+        {
+#pragma warning(suppress : 26490) // Don't use reinterpret_cast (type.1).
+            return std::launder(reinterpret_cast<POINT*>(this));

```

Comment:
first time i have ever seen somebody introduce a `launder` in the wild. Now, can you explain why it is necessary here?

Reply:
My understanding is:
C++ has `reinterpret_cast` but it's pretty much entirely useless.
You want to make an unsafe cast, because you know that the bitwise representation between both types is identical? Great. Because you can't. The language standard doesn't allow this and converting a `til::point` to `POINT` via `reinterpret_cast` is invalid no matter what.

`std::launder` is the next best tool in the language to say "I want what I want" even if it's purpose is not for this. The spec says given `std::launder(p)`: "All bytes of storage that would be reachable through the result are reachable through p". That's something I want: When you use that laundered pointer and make changes, I want those changes to be reachable through `this` as well. But my hope is not to be correct, but rather to not find any surprises. Under MSVC it shouldn't be much of an issue though, as this compiler doesn't handle strict aliasing at all.